src/Polynomials.jl

```diff
@@ -228,21 +224,21 @@ function setindex!(p::Poly, vs, idx::AbstractArray)
     [setindex!(p, v, i) for (i,v) in zip(idx, vs)]
     p
 end
-Base.eachindex{T}(p::Poly{T}) = 0:(length(p)-1)
+eachindex{T}(p::Poly{T}) = 0:(length(p)-1)
+
 
-    
 copy(p::Poly) = Poly(copy(p.a), p.var)
 
-zero{T}(p::Poly{T}) = Poly([zero(T)], p.var)
+zero{T}(p::Poly{T}) = Poly(T[], p.var)
```

Comment:
if you check the following line, you will see
```
 zero{T}(::Type{Poly{T}}) = Poly(T[])
```

I just thought we should keep the definitions consistent. If you are arguing that it might depend on the implementation detail, still keeping the definitions consistent might help.

I can see a slight difference from the constructor's perspective. `T[]` will by default branch to the empty array case, and `zeros(T,1)` will be triggered. `[zero(T)]`, in my opinion, brings a bit more overhead: 1) it will branch out to the non-empty case, 2) `findlast` gets trigerred, 3) `max` is called, 4) temporary is created thru slicing of the vector.

I know, compilers most likely optimize some of these problems. I just wanted to have a consistent definition to avoid confusion.